subsys/bluetooth/common/Kconfig

```diff
@@ -184,11 +184,34 @@ config BT_BUF_CMD_TX_SIZE
 
 config BT_BUF_CMD_TX_COUNT
 	int "Number of HCI command buffers"
-	default 2
-	range 2 64
+	default 2 if BT_HCI_ACL_FLOW_CONTROL
+	default 1
+	range 1 64
 	help
 	  Number of buffers available for outgoing HCI commands from the Host.
 
+	  HCI Controllers may not support Num_HCI_Command_Packets > 1, hence
+	  default to 1 when not enabling Controller to Host data flow control,
+	  BT_HCI_ACL_FLOW_CONTROL.
+
+	  Normal HCI commands follow the HCI command flow control using
+	  Num_HCI_Command_Packets return in HCI command complete and status.
+
+	  Host Number of Completed Packets command does not follow normal flow
+	  control of HCI commands and the Controller side HCI drivers that
+	  allocates HCI command buffers with K_NO_WAIT can end up running out
+	  of command buffers.
+
+	  When Controller to Host data flow control is enabled in the Host only
+	  build, ensure that BT_BUF_CMD_TX_COUNT is at least 2, to be able to
+	  send one normal HCI command and one Host Number of Completed Packets
+	  command.
```

Comment:
Can't this be enforced via Kconfig? Something like `range 2 64 if !BT_CTLR && BT_HCI_ACL_FLOW_CONTROL`?

Reply:
Yes, `depends on !BT_HCI_RAW || !HAS_BT_CTLR` ?

I want to keep BT_BUF_CMD_TX_COUNT only for Host only and Host+Controller build, but not for Controller-only build, as it does not make sense when the Controller only supports 1 Num HCI Command Packets + BT_BUF_ACL_RX_COUNT (for Host Num Completed Packets commands).... where do you think is the right place to define BT_BUF_CMD_TX_COUNT internally, so that it is reference today in hci_raw.c and in the Controller code?

Reply:
Can we keep it in the place here, but make it promptless and default to 1 for controller-only builds?

Reply:
Updated, lets see if CI is happy.